src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs

```diff
@@ -127,8 +136,32 @@ private void ProcessStartEvent(TRequest request, Uri requestUrl)
 				return;
 			}
 
-			var span = ExecutionSegmentCommon.StartSpanOnCurrentExecutionSegment(ApmAgent, $"{RequestGetMethod(request)} {requestUrl.Host}",
-				ApiConstants.TypeExternal, ApiConstants.SubtypeHttp, InstrumentationFlag.HttpClient, true);
+			var method = RequestGetMethod(request);
+			string HeaderGetter(string header) => RequestTryGetHeader(request, header, out var value) ? value : null;
+
+			ISpan span = null;
+
+			if (_realAgent != null)
+			{
+				using (var httpTracers = _realAgent.Components.GetHttpTracers())
+				{
+					foreach (var httpSpanTracer in httpTracers)
+					{
+						if (httpSpanTracer.IsMatch(method, requestUrl, HeaderGetter))
+						{
+							span = httpSpanTracer.StartSpan(ApmAgent, method, requestUrl, HeaderGetter);
+							if (span != null)
+								break;
+						}
+					}
+				}
+			}
+
+			if (span is null && StartHttpSpan)
```

Comment:
With this `span` can remain `null`, below we have some usages of the `span` variable and I think we should null-check those.

Reply:
Good point, will check